c/xlate.c

```diff
@@ -77,6 +80,48 @@ char* a2e(char *buffer, int len)
 }
 
 
+CharsetOracle *makeCharsetOracle(void){
+  CharsetOracle *oracle = (CharsetOracle*)safeMalloc(sizeof(CharsetOracle),"CharsetOracle");
+  memset(oracle,0,sizeof(CharsetOracle));
+  return oracle;
+}
+
+void freeCharsetOracle(CharsetOracle *oracle){
+  safeFree((char*)oracle,sizeof(CharsetOracle));
+}
+
+void charsetOracleDigest(CharsetOracle *oracle, char *s, int len){
```

Comment:
I think it'd be hard to change `int` to `size_t` in the future. So that change might be worth making right away. Not necessarily for this function, but just in general.

Reply:
Having discussed this with Joe, we'll do a general clean up later.